src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -543,6 +569,21 @@ def construct_payload(self):
             else:
                 set_managed_identity(self.cmd, self.get_argument_resource_group_name(), self.containerapp_def, user_assigned=[self.get_argument_registry_identity()])
 
+        if self.get_argument_source():
+            self.set_up_create_containerapp_if_source_or_repo()
+
+        if self.get_argument_repo():
+            container_app = self._get_containerapp_if_exists()
```

Comment:
Do we have to check container app exist first? the PUT operation should be idempotent. Why do we need to add image specifically?

Reply:
Executing the create command with `--repo` essentially makes two ARM calls. The container app is created first and the source control is set only after the container app is created. This means, when the container app is first created, the image would be defaulted to `mcr.microsoft.com/k8se/quickstart:latest` and the image name would change to `registry-server/containerapp-name:{github-sha} after the source control completes successfully. 

If the create call is executed multiple times however, we need to make sure that the image is not defaulted to `mcr.microsoft.com/k8se/quickstart:latest` if source control fails the subsequent times by setting the image to the previously deployed image.